### PR TITLE
Eth steth: fix Withdrawals: claimed amount is more than requested

### PR DESCRIPTION
**Pull request by sergeyWh1te**
Added: 
- Health check mechanism
- Wrappers for network and db findings
-  Removed from finding's description, Ts error object.
- Refactored **WithdrawalsSrv** from using Map for withdrawal_requests queue to sqlite table
- Tests for GateSealSrv
- Split up IETHprovider to IGateSealClient, IStethClient, IVaultClient, IWithdrawalsClient 

**Made audit for severity levels and their pictures**

### CODE REVIEW

File: ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts [hunk 1]
```diff
@@ -309,11 +276,10 @@ export class WithdrawalsSrv {
           name: `⚠️ Withdrawals: ${drainedStakeLimitRate.times(
             100,
           )}% of stake limit is drained and unfinalized queue is on par with drained stake limit`,
-          description: `Unfinalized queue: ${unfinalizedStETH.right
-            .div(ETH_DECIMALS)
-            .toFixed(2)} stETH\nDrained stake limit: ${drainedStakeLimit.div(ETH_DECIMALS).toFixed(2)} stETH`,
+          description: `Unfinalized queue: ${unfinalizedStETH.right.div(ETH_DECIMALS).toFixed(2)} stETH\n
+          Drained stake limit: ${drainedStakeLimit.div(ETH_DECIMALS).toFixed(2)} stETH`,
```
Comment (TheDZhon) [inline ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts:280]:
```suggestion
          Spent stake limit: ${spentStakeLimit.div(ETH_DECIMALS).toFixed(2)} stETH`,
```

File: ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts [hunk 2]
```diff
@@ -309,11 +276,10 @@ export class WithdrawalsSrv {
           name: `⚠️ Withdrawals: ${drainedStakeLimitRate.times(
```
Comment (TheDZhon) [inline ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts:276]:
```suggestion
          name: `⚠️ Withdrawals: ${spentStakeLimitRate.times(
```

File: ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts [hunk 3]
```diff
@@ -487,13 +467,11 @@ export class WithdrawalsSrv {
         out.push(
           Finding.fromObject({
             name: `⚠️ Withdrawals: ${unclaimedSizeRate.times(100).toFixed(2)}% of finalized requests are unclaimed`,
```
Comment (TheDZhon) [inline ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts:469]: Let's change: :warning:  → :information_source:

File: ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts [hunk 4]
```diff
@@ -487,13 +467,11 @@ export class WithdrawalsSrv {
...
-              .div(ETH_DECIMALS)
-              .toFixed(2)} stETH\nClaimed (for 2 weeks): ${claimedStETH
-              .div(ETH_DECIMALS)
-              .toFixed(2)} stETH\nTotal finalized: ${totalFinalizedSize.div(ETH_DECIMALS).toFixed(2)} stETH`,
+            description: `Unclaimed (for all time): ${unclaimedStETH.div(ETH_DECIMALS).toFixed(2)} stETH\n
+            Claimed (for 2 weeks): ${claimedStETH.div(ETH_DECIMALS).toFixed(2)} stETH\n
+            Total finalized: ${totalFinalizedSize.div(ETH_DECIMALS).toFixed(2)} stETH`,
             alertId: 'WITHDRAWALS-UNCLAIMED-REQUESTS',
-            severity: FindingSeverity.Info,
+            severity: FindingSeverity.Medium,
```
Comment (TheDZhon) [inline ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts:474]: I would suggest leaving 'info'

File: ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts [hunk 5]
```diff
@@ -505,33 +483,28 @@ export class WithdrawalsSrv {
...
+            `Error in ${WithdrawalsSrv.name}.${this.handleUnclaimedRequests.name}:490`,
+            `Could not call ethProvider.getBalance`,
+          ),
+        ]
       }
 
       if (unclaimedStETH.gt(withdrawalQueueBalance.right)) {
         out.push(
           Finding.fromObject({
             name: `⚠️ Withdrawals: unclaimed requests size is more than withdrawal queue balance`,
```
Comment (TheDZhon) [inline ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts:502]: ⚠️ ⇒ 🚨🚨🚨

It's critical and would need to activate GateSeal most likely

File: ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts [hunk 6]
```diff
@@ -505,33 +483,28 @@ export class WithdrawalsSrv {
...
-            description: `Unclaimed: ${unclaimedStETH
-              .div(ETH_DECIMALS)
-              .toFixed(2)} stETH\nWithdrawal queue balance: ${withdrawalQueueBalance.right
-              .div(ETH_DECIMALS)
-              .toFixed(2)} ETH\nDifference: ${unclaimedStETH.minus(withdrawalQueueBalance.right)} wei`,
+            description: `Unclaimed: ${unclaimedStETH.div(ETH_DECIMALS).toFixed(2)} stETH\n
+            Withdrawal queue balance: ${withdrawalQueueBalance.right.div(ETH_DECIMALS).toFixed(2)} ETH\n
+            Difference: ${unclaimedStETH.minus(withdrawalQueueBalance.right)} wei`,
             alertId: 'WITHDRAWALS-UNCLAIMED-REQUESTS-MORE-THAN-BALANCE',
-            severity: FindingSeverity.Critical,
```
Comment (TheDZhon) [inline ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts:534]: Should be critical

File: ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts [hunk 7]
```diff
@@ -717,27 +719,40 @@ export class WithdrawalsSrv {
...
             Finding.fromObject({
-              name: `⚠️Withdrawals: claimed amount is more than requested`,
-              description: `Request ID: ${etherscanNft(this.wdQueueContract.address, reqId)}\nClaimed: ${claimedAmount
-                .div(ETH_DECIMALS)
-                .toFixed(2)} ETH\nRequested: ${(curr.amount as BigNumber)
-                .div(ETH_DECIMALS)
-                .toFixed(2)} stETH\nDifference: ${claimedAmount.minus(
-                curr.amount as BigNumber,
-              )} wei\nOwner: ${etherscanAddress(event.args.owner)}\nReceiver: ${etherscanAddress(event.args.receiver)}`,
+              name: `⚠️ Withdrawals: claimed amount is more than requested`,
```
Comment (TheDZhon) [inline ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts:747]: 🚨🚨🚨

Should be critical (invariant violation)

File: ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts [hunk 8]
```diff
@@ -717,27 +719,40 @@ export class WithdrawalsSrv {
...
+              description: `Request ID: ${etherscanNft(this.withdrawalsQueueAddress, reqId)}\n
+              Claimed: ${claimedAmount.div(ETH_DECIMALS).toFixed(2)} ETH\n
+              Requested: ${withdrawalRequest.amountOfStETH.div(ETH_DECIMALS).toFixed(2)} stETH\n
+              Difference: ${claimedAmount.minus(withdrawalRequest.amountOfStETH)} wei\n
+              Owner: ${etherscanAddress(event.args.owner)}\n
+              Receiver: ${etherscanAddress(event.args.receiver)}`,
               alertId: 'WITHDRAWALS-CLAIMED-AMOUNT-MORE-THAN-REQUESTED',
-              severity: FindingSeverity.Critical,
+              severity: FindingSeverity.Medium,
```
Comment (TheDZhon) [inline ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts:755]: should be critical

**Review by TheDZhon — CHANGES_REQUESTED** [review]
:+1: Utterly cool, thank you so much!

I'd suggest tuning up some severities still before merging this one.

File: ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts (hunk 1 above)
Comment (sergeyWh1te) [reply ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts:280]: done

File: ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts (hunk 2 above)
Comment (sergeyWh1te) [reply ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts:276]: done

File: ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts (hunk 3 above)
Comment (sergeyWh1te) [reply ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts:469]: done

File: ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts (hunk 4 above)
Comment (sergeyWh1te) [reply ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts:474]: done

File: ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts (hunk 5 above)
Comment (sergeyWh1te) [reply ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts:502]: done

File: ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts (hunk 6 above)
Comment (sergeyWh1te) [reply ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts:534]: done

File: ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts (hunk 7 above)
Comment (sergeyWh1te) [reply ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts:747]: done

File: ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts (hunk 8 above)
Comment (sergeyWh1te) [reply ethereum-steth/src/services/withdrawals/Withdrawals.srv.ts:755]: done

**Review by TheDZhon — APPROVED** [review]
:+1:

**Pull request merged**